csrank/metrics.py

```diff
@@ -123,3 +123,79 @@ def topk_acc(y_true, y_pred):
         return acc
 
     return topk_acc
+
+
+def dcg_gain(grading, max_grade):
+    """Maps a grading (from 0 to `max_grade`) to its DCG "gain".
+
+    The gain is defined as in the Discounted Cumulative Gain (DCG)
+    metric. This is always a value in [0, 1] that can be interpreted as
+    a probability.
+    """
+    return (2 ** grading - 1) / (2 ** max_grade)
+
+
+def err(gradings, probability_mapping):
+    """Computes the Expected Reciprocal Rank.
+
+    ERR is the cascade metric with the reciprocal rank as the utility
+    function.
+
+    For more details see the original paper at
+    http://olivier.chapelle.cc/pub/err.pdf.
+    """
+    return cascade_metric(
+        gradings, probability_mapping, utility=lambda r: 1 / r
+    )
+
+
+def cascade_metric(gradings, probability_mapping, utility):
+    """Computes a cascade metric with a given utility function.
+
+    For more details see the original paper at
+    http://olivier.chapelle.cc/pub/err.pdf.
```

Comment:
This could be moved to the References block (see [here](https://github.com/kiudee/cs-ranking/blob/6c4c30cec89363b97d5c2c238cd0a77637b7a4ee/csrank/objectranking/cmp_net.py#L69-L71))

Reply:
I've made that change, but I'm not sure what citation style to use. For now I've used the MLA style (as given by google scholar).